src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -2490,6 +2495,16 @@ static int journal_file_read_tail_timestamp(sd_journal *j, JournalFile *f) {
         if (mo > rt) /* monotonic clock is further ahead than realtime? that's weird, refuse to use the data */
                 return -ENODATA;
 
+        if (offset == f->newest_entry_offset) {
+                /* Cached data and the current one should be equivalent. */
+                assert(sd_id128_equal(f->newest_boot_id, id));
+                assert(f->newest_monotonic_usec == mo);
+                assert(f->newest_realtime_usec == rt);
+                assert(sd_id128_equal(f->newest_machine_id, f->header->machine_id));
```

Comment:
this should not be an assert()... an rogue client might change these fields while we read them, making our assumptions. and boom we hit an assert(). it makes sense to validate that these are in order, but this should be an EBADMSG thing or so, not an assert()